EssentialCSharp.Web/Areas/Identity/Pages/Account/Register.cshtml.cs

```diff
@@ -0,0 +1,238 @@
+                    return Page();
+                }
+                IdentityResult result = await _UserManager.CreateAsync(user, Input.Password);
+
+                if (result.Succeeded)
+                {
+                    _Logger.LogInformation("User created a new account with password.");
+
+                    string userId = await _UserManager.GetUserIdAsync(user);
+                    string code = await _UserManager.GenerateEmailConfirmationTokenAsync(user);
+                    code = WebEncoders.Base64UrlEncode(Encoding.UTF8.GetBytes(code));
+                    string? callbackUrl = Url.Page(
+                        "/Account/ConfirmEmail",
+                        pageHandler: null,
+                        values: new { area = "Identity", userId = userId, code = code, returnUrl = returnUrl },
+                        protocol: Request.Scheme);
+
+                    if (callbackUrl is null)
+                    {
+                        ModelState.AddModelError(string.Empty, "Error: callback url unexpectedly null.");
+                        return Page();
+                    }
+                    if (Input.Email is null)
+                    {
+                        ModelState.AddModelError(string.Empty, "Error: Email may not be null.");
+                        return Page();
+                    }
+                    await _EmailSender.SendEmailAsync(Input.Email, "Confirm your email",
+                        $"Please confirm your account by <a href='{HtmlEncoder.Default.Encode(callbackUrl)}'>clicking here</a>.");
+
+                    if (_UserManager.Options.SignIn.RequireConfirmedAccount)
+                    {
+                        return RedirectToPage("RegisterConfirmation", new { email = Input.Email, returnUrl = returnUrl });
+                    }
+                    else
+                    {
+                        await _SignInManager.SignInAsync(user, isPersistent: false);
+                        return LocalRedirect(returnUrl);
+                    }
+                }
+                foreach (IdentityError error in result.Errors)
+                {
+                    ModelState.AddModelError(string.Empty, error.Description);
+                }
+            }
+        }
+        else
+        {
+            switch (response.ErrorCodes.Count)
+            {
+                case 0:
+                    throw new InvalidOperationException("The HCaptcha determined the passcode is not valid, and does not meet the security criteria");
+                case > 1:
+                    throw new InvalidOperationException("HCaptcha returned error codes: " + string.Join(", ", response.ErrorCodes));
+                default:
+                    {
+                        HCaptchaErrorDetails.TryGetValue(response.ErrorCodes.FirstOrDefault() ?? string.Empty, out HCaptchaErrorDetails? details);
+                        switch (details?.ErrorCode)
```

Comment:
Rather than this null check here, it would be better to put the TryGetValue in an if-statement. It should be decorated such that the analyzers knows on a true return `details` will not be null